### PR TITLE
[DQM] Various fixes/improvements for unit test

### PR DESCRIPTION
**Pull request by smuzaffar**
Updated unit tests so that they can be run from their own separate directory. Mostly 
- convert `<bin ...>` to `<test....>`
- use full path e.g `CMSSW_BASE/src/...` instead of `./src/`

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

File: Validation/RecoTrack/test/BuildFile.xml [hunk 1]
```diff
@@ -1,4 +1 @@
-<bin name="testMakeTrackValidationPlots" file="TestDriver.cpp">
-  <flags TEST_RUNNER_ARGS="/bin/bash Validation/RecoTrack/test test_makeTrackValidationPlots.sh"/>
-  <use name="FWCore/Utilities"/>
-</bin>
+<test name="testMakeTrackValidationPlots" command="test_makeTrackValidationPlots.sh"/>
```
Comment (mmusich) [inline Validation/RecoTrack/test/BuildFile.xml:1]: why is this better?

Comment (smuzaffar) [reply Validation/RecoTrack/test/BuildFile.xml:1]: You do not need an extra `TestDriver.cpp` file which also means that scram will not compile/build an extra executable.

Comment (smuzaffar) [reply Validation/RecoTrack/test/BuildFile.xml:1]: I should have deleted `TestDriver.cpp` too

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
@cms-sw/dqm-l2 can you please review this?

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**